containers/podman-desktop
# feat: move podman extension to packages/extension folder and add dev and runtime support for it

### What does this PR do?

This PR adds support for extensions with API, when extension folder have multiple packages api and extension itself under extensions/ext-name/packages folder.

podman extension structure changed to new one by moving extensions/podman to extensions/podman/packages/extension folder.

### Screenshot / video of UI



### What issues does this PR fix or reference?

This PR is extracted from #8045 to simplify review.

### How to test this PR?

1. Pull the branch
2. `git status` should not show any folders form `extensions/podman/` as untracked
3. `git -C extensions/podman/ clean -dxf` to remove ignored files left after checkout
1. pnpm typecheck:extensions:podman
4. pnpm lint:check
5. pnpm build:extensions:podman
6. pnpm test:extensions:podman
7. pnpm watch
8. build inastaller and run it to test in production environment or pull the branch https://github.com/dgolovin/podman-desktop/tree/allow-to-run-in-electron with rebase on top of this one. run `pnpm build` && `npx electron .`
